City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-2233: E2E, Moved various variables to the .env file

# [AU-2233](https://helsinkisolutionoffice.atlassian.net/browse/AU-2233)

## What was done

This pull request implements the following things:

- Moved the `TEST_SSN` and `TEST_USER_UUID` variables from `test_data.ts` to the `.env` file.

- Moved the `TEST_IBAN` variable from `test_data.ts` to the profile input data implementation in `profile_input_data.ts`.

- Moved relevant `attachment variables` from `helpers.ts` to `attachment_data.ts`. Also moved all the attachment files to a new `/attachments` directory.

- Implemented the `checkEnvVariables()` function in `env_helpers.ts`.  This function is being called in `init.setup.ts`, and it makes sure that all the needed `.env` variables are set before any tests are run.

#### Other minor changes

- Cleaned up import statements in various files.

- Changed the `CREATE_PROFILE` check from `==='true'` to `==='TRUE'` (basically made the variable uppercase so that it matches with the other `.env` variables.

## What has changed

- The `TEST_SSN` and `TEST_USER_UUID` should now be set in the `.env` file.

- Any new attachment files should be added to the `/attachments` directory. The file should also be added to the `AttachmentFiles` implementation in `attachment_data.ts`.

- When you want to use an attachment in test data, you can simply add the following import statement instead of having to import every attachment separately: `import {ATTACHMENTS} from "../attachment_data";`

- The following variables always need to be set in the `.env` file for the tests to execute:
  -  ATV_BASE_URL
  -  TEST_ATV_URL
  -  ATV_API_KEY
  -  APP_ENV
  -  TEST_USER_SSN
  -  TEST_USER_UUID


## How to install

Make sure your instance is up and running on correct branch.
* `git checkout feature/AU-2233-move-variables-to-env`
* `make fresh`
* `make drush-cr`

## How to test
- Make sure your `.env` file looks something like this. Swap the `ENTER_VALUE` values with what you were using previously.

```
# ATV SETUP.
ATV_BASE_URL="ENTER_VALUE"
TEST_ATV_URL="ENTER_VALUE"
ATV_API_KEY="ENTER_VALUE"
APP_ENV="ENTER_VALUE"

# Enable debug mode. Primarily used for logging messages.
APP_DEBUG="TRUE"

# Set the disabled form variants.
DISABLED_FORM_VARIANTS="success"

# Force profile creation.
CREATE_PROFILE="TRUE"

# Testa user data (SSN (SOTU) and UUID).
TEST_USER_SSN="ENTER_VALUE"
TEST_USER_UUID="ENTER_VALUE"
```

- Delete the `user.json` file in the `.auth` directory if you changed the `SSN` or `UUID` you typically use.

- Run a few tests with `make test-pw` and make sure that they work correctly.

- Remove a critical `.env` variable like `ATV_API_KEY` or `TEST_USER_SSN` from the `.env` file and run the tests again. Make sure that an error is thrown and that no tests are executed.

[AU-2233]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
